### ISSUE TITLE
updateModalImages() in show$ does not work

### ISSUE THREAD

**Issue by yarozar** [Ks89/angular-modal-gallery#265]
When running updateModalImages() inside subscriptions of `show$()` (or `firstImage$`, `lastImage$`) you still get an image from before updateModalImages() array shown, and only next navigation clicks get you to the refreshed images array.

In manual it says that `show$` is fired BEFORE image visible so I expected to override current image with updateModalImages() ...

Also, from this as a workaround comes a question if it is possible to programmatically change `currentImage` in already opened gallery? Did not find anything in docs.

Thanks!

p.s.: your gallery looks and works great, thank you very much for making it available for everyone!

**Comment by Ks89** [comment]
> In manual it says that show$ is fired BEFORE image visible

Are you saying that show$ is fired AFTER instead of BEFORE? 


> if it is possible to programmatically change currentImage in already opened gallery

At the moment, this is not possible.

**Comment by Ks89** [comment]
Mmm ok now I understand. Yes it Is asynchronous. I'll try to do some experiments.

**Comment by Ks89** [comment]
I made some experiment but I don't know which workaround could work to implement this.
I modified the code of the library to emit the event after a delay and in this way it works, but it isn't the right way to implement it.

To add this feature we need to modify the library to support pagination, but this is a huge change.
We need to track exactly which features to implement, for instance:
- add a way to change currentImage via service (but this should be done only choosing an image from the images array)
- think a way to add pagination

**Comment by Ks89** [comment]
Yes, but it's a workaround, I cannot add this feature to a library used by many other people. This could cause other issues or misuses.
The right way to implement this is adding pagination as a real feature to the library.

However I still don't understand why you don't append images to the array instead of replace them.
Let say that I have 50 images and I receive these in 5 pages of 10 images.
When the async service that retrieve the images has some data, it adds to the array automatically, without replacing anything in the array. The only thing you need to know is when you can call the service to obtain the next page (for instance you reached 50% of the current page).
I never tried to implement this, but if you have a working example to simulate your scenario I can try locally to do some experiments. If you can, please create a new public repo based on the official `example/angular-cli-16`.
